### ISSUE TITLE
Tests incorrectly failing

### ISSUE THREAD

**Issue by alan-strohm** [ianthehenry/judge#9]
With Judge v2.8.0, repeatedly running judge -a on the following code always fails.

```
(import judge)

(defn bad-fn []
  (def r @{})
  (for n 0 3
    (put-in r [[n (inc n)] [n (inc n)]] true))
  r)

(judge/test
  (bad-fn)
  @{[0 1] @{[0 1] true}
    [1 2] @{[1 2] true}
    [2 3] @{[2 3] true}})
```

This is the smallest reproduction I could find.  With only 2 entries, it works fine.  Without the call to inc, it works fine.  Without tuple keys, it works fine.  With tuples of length 1, it works fine.

**Comment by ianthehenry** [comment]
Hey thanks! Great repro. Neat bug.

```
repl:1:> (deep= @{[0 1] 1 [1 2] 2} @{'[0 1] 1 '[1 2] 2})
true
repl:2:> (deep= @{[0 1] 1 [1 2] 2 [2 3] 3} @{'[0 1] 1 '[1 2] 2 '[2 3] 3})
false
```

The problem here is that, since Janet 1.27.0, bracketed tuples are no longer equal to regular tuples. But they are still `deep=` to them, which I think is pretty much a bug, but a bug that has worked to our advantage up until now. But the different types of tuples also hash to different values, which means that hash tables with bracketed tuple keys may be ordered differently than hash tables with regular tuple keys, if you get unlucky. Or they could be the same -- which is why this repro is so fragile.

`judge` has always printed all tuples without source maps as bracketed tuples, which means that they could be read in in a different order. I think the fix for this is won't be too bad -- we just need to do the bracketed tuple conversion a little bit earlier. Although we have a weird problem if you have a table with otherwise identical bracketed and unbracketed tuple keys... will have to make sure to handle that.

**Comment by ianthehenry** [comment]
I believe that this is fixed in 2.8.1, could you give that a try?

**Comment by alan-strohm** [comment]
2.8.1 does indeed fix the problem. Thanks for the quick fix!
